WW-Digital/reactive-kinesis
# Fix #25 by providing an Akka Stream Source graph stage for Kinesis.

This graph stage provides an akka source graph based on the actor model provided by the reactive kinesis consumer.

If a consumer is configured, an akka stream Source can be obtained via a simple `Kinesis.source("consumer-name")`.
Every message that flows through the stream needs to be committed explicitly.

An integration test is added that relies on [localstack](https://github.com/localstack/localstack) running on the same node (I did not add the localstack startup logic as part of this PR).

Steps to run the test:
1) Set this env var, since the localstack does not support CBOR: `export AWS_CBOR_DISABLE=true`
2) Start localstack (I use the docker-compose with: `docker-compose -f localstack.yml up`) in default configuration (Kinesis on port 4568, Dynamo on port 4569)
3) ```sbt it:test```

It was not as easy to come up with a good integration test, since a Kinesis Source usually never finishes. To have a finite test, I use `Flow.take` which ends the stream after a defined amount of entries. The interplay of different readers/shard consumers could be different between runs, but the test expectation should be met.
